scripts/inference/ft/mpt_ckpt_to_ft.py

```diff
@@ -0,0 +1,294 @@
+        data = param.detach().cpu().numpy().astype(np_weight_data_type)
+        if name.find('weight') == -1 and name.find('bias') == -1:
+            print(f'found a parameter name that is not handled: {name}')
+            continue
+        if name == 'transformer.wpe.weight':
+            assert data.shape == (
+                hf_config['max_seq_len'],
+                hf_config['d_model']), f'unexpected dim for {name}'
+            data.tofile(save_dir + 'model.wpe.bin')
+        elif name == 'transformer.wte.weight':
+            assert data.shape == (
+                hf_config['vocab_size'],
+                hf_config['d_model']), f'unexpected dim for {name}'
+            data.tofile(save_dir + 'model.wte.bin')
+        elif name == 'transformer.norm_f.bias':
+            assert data.shape == (
+                hf_config['d_model'],), f'unexpected dim for {name}'
+            data.tofile(save_dir + 'model.final_layernorm.bias.bin')
+        elif name == 'transformer.norm_f.weight':
+            assert data.shape == (
+                hf_config['d_model'],), f'unexpected dim for {name}'
+            save_path = save_dir + 'model.final_layernorm.weight.bin'
+            data.tofile(save_path)
+            if hf_config['no_bias']:
+                write_empty_bias(name, save_path, data.shape[-1])
+        elif name == 'transformer.lm_head.weight':
+            data.tofile(save_dir + 'model.lm_head.weight.bin')
+        else:
+            for mpt_pattern, ft_pattern in param_remapping.items():
+                if name.find(mpt_pattern) != -1:
+                    new_name = name.replace('transformer.blocks.',
+                                            'layers.').replace(
+                                                mpt_pattern, ft_pattern)
+                    pool.starmap(
+                        split_and_convert_process,
+                        [(0, save_dir, factor, new_name, hf_config, data)],
+                    )
+
+    pool.close()
+    pool.join()
+
+
+if __name__ == '__main__':
+    parser = argparse.ArgumentParser(
+        formatter_class=argparse.RawTextHelpFormatter)
+    parser.add_argument('-save_dir',
+                        '-o',
+                        type=str,
+                        help='Dir to save converted checkpoint',
+                        required=True)
+    parser.add_argument(
+        '-name_or_dir',
+        '-i',
+        type=str,
+        help=
+        'HF hub Model name (e.g., mosaicml/mpt-7b) or local dir path to load checkpoint from',
+        required=True)
+    parser.add_argument('-infer_gpu_num',
+                        '-i_g',
+                        type=int,
+                        help='How many gpus for inference?',
+                        required=True)
+    parser.add_argument(
+        '-processes',
+        '-p',
+        type=int,
+        help='How many processes to spawn for conversion? (default: 4)',
+        default=4)
+    parser.add_argument('-weight_data_type',
```

Comment:
What does this arg mean? Is it the data type of the original weights? or the desired output data type? (and add the answer to a `help`)

Reply:
Added an explanation.